### ISSUE TITLE
Skip JavaScript template tokens like {{name}} while rendering

### ISSUE THREAD

**Issue by scratcher28** [bcosca/fatfree#122]
Hi.

I've a JavaScript template in my F3 template which contains tokens like {{name}}
So, F3's renderer outputs:

500 Internal Server Error
Use of undefined constant name - assumed 'name'

Why? It's not a F3's template token, right?
How can I fix it?

**Comment by jsheperd** [comment]
Angular is ready to use custom template signs.
https://docs.angularjs.org/api/ng/provider/$interpolateProvider

**Comment by bojanra** [comment]
Why not make the fatfree template delimiters more strict!
Typically everything between braces `{{ }}` is taken as templating.
I had the same problem and extended the starting braces '{{' with a following space. 
e.g.
'{{ @title'}} is normally rendered, but
'{{@title}}' is ignored.

For this you only need to correct the regular expressions inside the template engine in base.php

**Comment by sn0opy** [comment]
@bojanra that's actually not really a good idea, since that's something people tend to do depending on their preferences. Some like to have a space in there, others don't. 

Beside that it might break a lot of peoples' code.

**Comment by bojanra** [comment]
@sn0opy I'm aware of this. 

I got the idea from a German f3 template documentation, where they write that spaces after and before the braces are explicitly required [PHP F3: Templates](https://easysolutionsit.de/artikel/php-f3-templates.html).
Nevertheless in the samples in the f3 documentation always spaces are used. 

We started to use [handlebarsjs](http://handlebarsjs.com/) and this was a good solution. We checked all existing template usage and added missing spaces.
